src/jit/morph.cpp

```diff
@@ -11496,7 +11501,16 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                         }
                     }
                 }
-                fgMorphRecognizeBoxNullable(tree);
+
+                __fallthrough;
+
+            case GT_GT:
+
+                if ((oper != GT_GT) || ((tree->gtFlags & GTF_UNSIGNED) != 0))
```

Comment:
We allow unsigned GT_GT as well as GT_EQ and GT_NE. Other comparison operators are not allowed on reference types according to ECMA-335.